api4/file.go

```diff
@@ -89,12 +91,17 @@ func uploadFile(c *Context, w http.ResponseWriter, r *http.Request) {
 			c.SetPermissionError(model.PERMISSION_UPLOAD_FILE)
 			return
 		}
-
+		body, err := ioutil.ReadAll(r.Body)
```

Comment:
This still results in putting the entire file into memory instead of streaming it, right?

Reply:
After reading the golang http request implementation, the request.ParseMultipartForm has loaded the file to be uploaded into memory (or memory and disk temporary files) when uploading the file. I guess we can do as little as possible to duplicate the memory. By the way, have we considered avoiding uploading duplicate files? For example, according to the md5 value to determine whether the file needs to be uploaded, the other is to adjust the file ownership, use, view and other permissions work.
Thank you for pointing out that there are bugs in the code, I have fixed and submitted again

Reply:
As we discussed earlier, the implementation of request.ParseMultipartForm is based on the size of the uploaded file to determine whether the uploaded file is read into memory at a time or stored in a temporary disk file. This place we cannot change. The improvement that I am making is to avoid multiple copies of the file from the request. ParseMultipartForm file in our MM business process, including file information, image file processing, etc. According to the current improvement the worst case is Copy once again When using request.body to upload files, memory replication is no longer performed in other cases